pin/pin.go

```diff
@@ -97,16 +102,19 @@ func (p *pinner) Pin(node *mdag.Node, recurse bool) error {
 			return err
 		}
 	} else {
+		if p.recursePin.HasKey(k) {
+			return errors.New("Key already pinned recursively.")
+		}
 		p.directPin.AddBlock(k)
 	}
 	return nil
 }
 
-// Unpin a given key with optional recursive unpinning
-func (p *pinner) Unpin(k util.Key, recurse bool) error {
+// Unpin a given key
+func (p *pinner) Unpin(k util.Key) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
-	if recurse {
+	if p.recursePin.HasKey(k) {
```

Comment:
Oh, i must have assumed the opposite after our discussion. The reason this makes sense to me is because:

```
> ipfs pin add -r h1
Pinned h1
> ipfs pin add h1
Error: pin: Key already pinned recursively.
```
A key can only be pinned once, and therefore, makes sense (to me at least) that a single rm operation should unpin it.